ts/parts/SVGLabel.ts

```diff
@@ -0,0 +1,548 @@
+/* *
+ *
+ *  (c) 2010-2020 Torstein Honsi
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+import type SVGRenderer from './SVGRenderer';
+import SVGElement from './SVGElement.js';
+import U from './Utilities.js';
+const {
+    defined,
+    extend,
+    isNumber,
+    merge,
+    removeEvent
+} = U;
+
+/**
+ * SVG label to render text.
+ * @private
+ * @class
+ * @name Highcharts.SVGLabel
+ * @augments Highcharts.SVGElement
+ */
+class SVGLabel extends SVGElement {
+
+    /* *
+     *
+     *  Static Properties
+     *
+     * */
+
+    public static readonly emptyBBox: Highcharts.BBoxObject = { width: 0, height: 0, x: 0, y: 0 };
+
+    /* *
+     *
+     *  Constructors
+     *
+     * */
+
+    public constructor(
+        renderer: SVGRenderer,
+        str: string,
+        x: number,
+        y?: number,
+        shape?: Highcharts.SymbolKeyValue | string,
+        anchorX?: number,
+        anchorY?: number,
+        useHTML?: boolean,
+        baseline?: boolean,
+        className?: string
+    ) {
+        super();
+        this.init(renderer, 'g');
+
+        this.textStr = str;
+        this.x = x;
+        this.y = y;
+        this.anchorX = anchorX;
+        this.anchorY = anchorY;
+        this.baseline = baseline;
+        this.className = className;
+
+        if (className !== 'button') {
+            this.addClass('highcharts-label');
+        }
+        if (className) {
+            this.addClass('highcharts-' + className);
+        }
+
+        this.text = renderer.text('', 0, 0, useHTML)
+            .attr({
+                zIndex: 1
+            });
+
+        // Validate the shape argument
+        let hasBGImage;
+        if (typeof shape === 'string') {
+            hasBGImage = /^url\((.*?)\)$/.test(shape);
+            if (this.renderer.symbol(shape) || hasBGImage) {
+                this.symbolKey = shape;
+            }
+        }
+
+        this.bBox = SVGLabel.emptyBBox;
+        this.padding = 3;
+        this.paddingLeft = 0;
+        this.baselineOffset = 0;
+        this.needsBox = renderer.styledMode || hasBGImage;
+        this.deferredAttr = {};
+        this.alignFactor = 0;
+
+    }
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    /**
+     * @private
+     */
+    public kind: string = 'label';
+
+    /**
+     * For labels, these CSS properties are applied to the `text` node directly.
+     *
+     * @private
+     * @name Highcharts.SVGLabel#textProps
+     * @type {Array<string>}
+     */
+    public textProps: Array<string> = [
+        'color', 'cursor', 'direction', 'fontFamily', 'fontSize', 'fontStyle',
+        'fontWeight', 'lineHeight', 'textAlign', 'textDecoration',
+        'textOutline', 'textOverflow', 'width'
+    ]
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    public alignSetter(value: 'left' | 'center' | 'right'): void {
```

Comment:
How about using [`H.AlignValue`](https://api.highcharts.com/class-reference/Highcharts.html#.AlignValue) instead of `string|string|string` ?